File: app/src/main/java/br/edu/ifce/engcomp/francis/radarpolitico/models/Votacao.java
package br.edu.ifce.engcomp.francis.radarpolitico.models;

import java.util.ArrayList;

/**
 * Created by francisco on 12/03/16.
 */
public class Votacao {
    private String resumo;

    private String data;
    private String hora;

    private String objetoVotacao;
    private String codigoSessao;

    private int totalVotosSim;
    private int totalVotosNao;
    private int totalVotosAbstencao;
    private int totalVotosObstrucao;
    private int totalVotosSessao;

    private ArrayList<Voto> votos;

    public Votacao() {
        this.votos = new ArrayList<>();
    }

    public ArrayList<Voto> getVotos() {
        return votos;
    }

    public void setVotos(ArrayList<Voto> votos) {
        this.votos = votos;
    }

    public String getResumo() {
        return resumo;
    }

    public void setResumo(String resumo) {
        this.resumo = resumo;
    }

    public String getData() {
        return data;
    }

    public void setData(String data) {
        this.data = data;
    }

    public String getHora() {
        return hora;
    }

    public void setHora(String hora) {
        this.hora = hora;
    }

    public String getObjetoVotacao() {
        return objetoVotacao;
    }

    public void setObjetoVotacao(String objetoVotacao) {
        this.objetoVotacao = objetoVotacao;
    }

    public String getCodigoSessao() {
        return codigoSessao;
    }

    public void setCodigoSessao(String codigoSessao) {
        this.codigoSessao = codigoSessao;
    }

    public int getTotalVotosSim() {
        return totalVotosSim;
    }

    public void setTotalVotosSim(int totalVotosSim) {
        this.totalVotosSim = totalVotosSim;
    }

    public int getTotalVotosNao() {
        return totalVotosNao;
    }

    public void setTotalVotosNao(int totalVotosNao) {
        this.totalVotosNao = totalVotosNao;
    }

    public int getTotalVotosAbstencao() {
        return totalVotosAbstencao;
    }

    public void setTotalVotosAbstencao(int totalVotosAbstencao) {
        this.totalVotosAbstencao = totalVotosAbstencao;
    }

    public int getTotalVotosObstrucao() {
        return totalVotosObstrucao;
    }

    public void setTotalVotosObstrucao(int totalVotosObstrucao) {
        this.totalVotosObstrucao = totalVotosObstrucao;
    }

    public int getTotalVotosSessao() {
        return totalVotosSessao;
    }

    public void setTotalVotosSessao(int totalVotosSessao) {
        this.totalVotosSessao = totalVotosSessao;
    }
}

File: app/src/main/kotlin/br/edu/ifce/engcomp/francis/radarpolitico/models/Votacao.kt
package br.edu.ifce.engcomp.francis.radarpolitico.models

import android.os.Parcel
import android.os.Parcelable
import java.util.*

/**
 * Created by francisco on 12/03/16.
 */
data class Votacao(
    var resumo: String? = "",

    var data: String? = "",
    var hora: String? = "",

    var objetoVotacao: String? = "",
    var codigoSessao:  String? = "",

    var totalVotosSim: String = "",
    var totalVotosNao: String = "",
    var totalVotosAbstencao: String = "",
    var totalVotosObstrucao: String = "",
    var totalVotosSessao: String =  "",

    var votos: ArrayList<Voto> = ArrayList<Voto>()) : Parcelable {

    constructor(source: Parcel): this(
            source.readSerializable() as String?,
            source.readSerializable() as String?,
            source.readSerializable() as String?,
            source.readSerializable() as String?,
            source.readSerializable() as String?,
            source.readString(), source.readString(),
            source.readString(), source.readString(),
            source.readString())

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel?, flags: Int) {
        dest?.writeSerializable(resumo)
        dest?.writeSerializable(data)
        dest?.writeSerializable(hora)
        dest?.writeSerializable(objetoVotacao)
        dest?.writeSerializable(codigoSessao)
        dest?.writeString(totalVotosSim)
        dest?.writeString(totalVotosNao)
        dest?.writeString(totalVotosAbstencao)
        dest?.writeString(totalVotosObstrucao)
        dest?.writeString(totalVotosSessao)
    }

    companion object {
        @JvmField final val CREATOR: Parcelable.Creator<Votacao> = object : Parcelable.Creator<Votacao> {
            override fun createFromParcel(source: Parcel): Votacao {
                return Votacao(source)
            }

            override fun newArray(size: Int): Array<Votacao?> {
                return arrayOfNulls(size)
            }
        }
    }

    fun merge(vt: Votacao) {
        this.resumo = if (vt.resumo == null) this.resumo else vt.resumo

        this.data = if (vt.data == null) this.data else vt.data
        this.hora = if (vt.hora == null) this.hora else vt.hora

        this.objetoVotacao = if (vt.objetoVotacao == null) this.objetoVotacao else vt.objetoVotacao
        this.codigoSessao = if (vt.codigoSessao == null) this.codigoSessao else vt.codigoSessao
        this.totalVotosSim = if (vt.totalVotosSim == null) this.totalVotosSim else vt.totalVotosSim
        this.totalVotosNao = if (vt.totalVotosNao == null) this.totalVotosNao else vt.totalVotosNao

        this.totalVotosAbstencao = if (vt.totalVotosAbstencao == null) this.totalVotosAbstencao else vt.totalVotosAbstencao
        this.totalVotosObstrucao = if (vt.totalVotosObstrucao == null) this.totalVotosObstrucao else vt.totalVotosObstrucao
        this.totalVotosSessao = if (vt.totalVotosSessao == null) this.totalVotosSessao else vt.totalVotosSessao

    }
}
